test/set_sharding_metadata_test.lua

```diff
@@ -420,10 +424,11 @@ function g.test_ddl_user_sharding_func_cdata_call()
         } test_set_struct_t;
     ]]
 
-    local test_set_struct = ffi.metatype('test_set_struct_t', {
+    ffi.metatype('test_set_struct_t', {
         __call = function(_, key) return key end
     })
-    rawset(_G, user_sharding_func_name, test_set_struct())
+    local test_set_struct = ffi.new('test_set_struct_t')
+    rawset(_G, user_sharding_func_name, test_set_struct)
```

Comment:
Can you please highlight for me what is wrong with this test and what is the difference?

Old version:

```lua
tarantool> ffi = require('ffi')
---
...

tarantool> ffi.cdef[[
        typedef struct
        {
            int data;
        } test_set_struct_t;
    ]]
---
...

tarantool> test_set_struct = ffi.metatype('test_set_struct_t', {
        __call = function(_, key) return key end
    })
---
...

tarantool> user_sharding_func = test_set_struct()
---
...

tarantool> user_sharding_func(2)
---
- 2
...

```

New version:

```lua
tarantool> ffi = require('ffi')
---
...

tarantool> ffi.cdef[[
        typedef struct
        {
            int data;
        } test_set_struct_t;
    ]]
---
...

tarantool> ffi.metatype('test_set_struct_t', {
         __call = function(_, key) return key end
     })
---
- ctype<struct 1071>
...

tarantool> test_set_struct = ffi.new('test_set_struct_t')
---
...

tarantool> user_sharding_func = test_set_struct
---
...

tarantool> user_sharding_func(2)
---
- 2
...
```

Reply:
> Can you please highlight for me what is wrong with this test and what is the difference?

Nothing wrong, actually. I found a new version of assigning metatable to cdata more clear and decide to rewrite similar code in your tests too. If it makes you sad, I can revert a patch.

Reply:
> Nothing wrong, actually.

But your commit message tells us: 'Fix tests for <...>'. Something that is not broken cannot be fixed. Refactored, cleaned up, clarified, simplified, revisited, transformed, rewritten — okay.

Reply:
Updated commit message and moved commit on top of the branch to order changes in a logical way.